crates/query-engine/execution/src/mutation.rs

```diff
@@ -0,0 +1,287 @@
+                    row_result_json.insert(
+                        col.name().to_owned(),
+                        date_time_str.map(|d| serde_json::Value::String(d.to_string())),
+                    );
+                }
+
+                _ => {
+                    return Err(Error::Mutation(MutationError::NativeMutation {
+                        column_name: col.name().to_string(),
+                        column_type: col.column_type(),
+                        error: NativeMutationResponseParseError::UnknownType(col.column_type()),
+                    }))
+                }
+            }
+        }
+
+        db_results_json.push(row_result_json);
+    }
+
+    serde_json::to_string(&db_results_json)
+        .map_err(|e| Error::Mutation(MutationError::JSONSerializationError(e)))
+}
+
+pub async fn execute_mutations(
+    mssql_pool: &bb8::Pool<bb8_tiberius::ConnectionManager>,
+    metrics: &metrics::Metrics,
+    plan: sql::execution_plan::MutationExecutionPlan,
+) -> Result<Bytes, Error> {
+    let acquisition_timer = metrics.time_connection_acquisition_wait();
+    let connection_result = mssql_pool
+        .get()
+        .instrument(info_span!("Acquire connection"))
+        .await
+        .map_err(Error::ConnectionPool);
+    let mut connection = acquisition_timer.complete_with(connection_result)?;
+
+    let query_timer = metrics.time_mutation_execution();
+    let mutation_response = rollback_on_exception(
+        execute_mutations_transaction(plan, &mut connection).await,
+        &mut connection,
+    )
+    .await;
+    query_timer.complete_with(mutation_response)
+}
+
+async fn execute_mutation(
+    connection: &mut bb8::PooledConnection<'_, bb8_tiberius::ConnectionManager>,
+    mutation_plan: MutationOperationExecutionPlan,
+    buffer: &mut (impl BufMut + Send),
+) -> Result<(), Error> {
+    match mutation_plan {
+        MutationOperationExecutionPlan::NativeMutation(native_mutation_plan) => {
+            execute_native_mutation(connection, native_mutation_plan, buffer).await
+        }
+    }
+}
+
+/// Execute the mutation query.
+async fn execute_native_mutation(
+    connection: &mut bb8::PooledConnection<'_, bb8_tiberius::ConnectionManager>,
+    native_mutation_plan: NativeMutationOperationExecutionPlan,
+    buffer: &mut (impl BufMut + Send),
+) -> Result<(), Error> {
+    let mutation_query = &native_mutation_plan.mutation_sql_query;
+
+    // User provided native mutation query.
+    let mut mssql_query = tiberius::Query::new(mutation_query.sql.as_str());
+
+    // bind parameters....
+    for param in mutation_query.params.clone().into_iter() {
+        match param {
+            sql::string::Param::String(string) => {
+                mssql_query.bind(string);
+            }
+            // Variables are not used with mutations.
+            sql::string::Param::Variable(_) => {}
+        }
+    }
+
+    // go!
+    let stream = mssql_query
+        .query(connection)
```

Comment:
why are we using `query` here instead of `execute`? According to the documentation, `query` is "Useful for `SELECT` statements" while `execute` is "Useful for `INSERT`, `UPDATE` and `DELETE` statements". Is it because `execute` only returns the number of affected rows, and not the rows themselves?

Reply:
> Is it because execute only returns the number of affected rows, and not the rows themselves?

Exactly!